libexec/python/docker/api.py

```diff
@@ -172,12 +174,17 @@ def get_token_url(self, challenge, expires_in, sort_query_params=False):
             query_fragment=query_fragment
         )
 
-    def update_token(self, response=None, auth=None, expires_in=9000):
+    def update_token(self, response=None, auth=None, expires_in=300):
         '''update_token uses HTTP basic authentication to get a token for
         Docker registry API V2 operations. We get here if a 401 is
         returned for a request.
         https://docs.docker.com/registry/spec/auth/token/
         '''
+
+        if self.tokenExpires and self.tokenExpires > (time.time() - 5):
```

Comment:
I don't think it will ever take close to 5 seconds between the check and making the request. However, if it does we are already covered. The logic in the main get method will call update_token and try again if it receives a 401 Unauthorized response for the request.